### ISSUE TITLE
Resolution supported

### ISSUE THREAD

**Issue by napgab** [asciidisco/plugin.video.netflix#27]
Which kind of resolution are supported?
  ##For now, i know that full hd and ultra hd on pc are avaiable only on windows through edge. 
with this plugin, can i see (at least) 1080p content on linux?

**Comment by asciidisco** [comment]
There´s no absolute answer to this question, as this is always hardware dependent when using Widevine to decrypt & decode the video, as it can be done through software decoding only.
Rule of thumb, if your CPU is powerful enough & enough RAM is available, 1080 is possible, regardless of the OS used.

What do I mean with powerful enough? At the moment an Odroid C2 (1.5 GHz Quad Core with 2gb RAM) is not able to do more than 720 reliably. But this is subject to change, as the `inputstream.adaptive` dependency (which basically is the bridge between Widevine & Kodi) is still work in progress. It will eventually be possible to have 1080 on the RPi 3 sometime in the future, but this is based on educated guesses, not on hard facts.

Regarding 4k support: You probably need to be a wizard to know if it will someday work or not. As Widevine does not (seem) to support it at the moment. But Netflix [announced recently](http://techblog.netflix.com/2017/03/update-on-html5-video-for-netflix.html) that they´re working with browser vendors (foremost Google, when it comes down to Widevine support) to get 4k working in the Browser. How this will be done, no Idea. I´d guess you need some kind of hardware acceleration (e.g. hardware decoding) to make it work...

TL:DR; If you´ve got enough CPU power & RAM, 1080 work, 4k doesn't & there´s hope to make 1080 work on "low power" machines like the RPi 3.

**Comment by napgab** [comment]
there is a way to check live streaming resolution?

**Comment by asciidisco** [comment]
@napgab You can press the "o" key on your keyboard in Kodi when a video is running, that´ll give you all the information inlcuding the resolution.
Also, you can set the max & min resolution in the inputstream.adaptive settings

**Comment by Uukrull** [comment]
The problem is most of the movies are streamed at 480p with a bitrate under 2 Mbps. With the Shield app, I can see the same movie at 1080p with a 7 Mbps bitrate. Would it be possible to "simulate" an app instead of a web browser?

**Comment by montvid** [comment]
See here for 1080p support on linux https://github.com/truedread/netflix-1080p

**Comment by montvid** [comment]
With the chrome plugin I see that I need to use super netflix to enforce a better bitrate because a movie starts with 1080p and in the middle starts caching 720p and goes to 720p. But yes, some movies don't have 1080p just 720p or less like "Friends" I think it is just the way Netflix ripped them. Thanks for the great plugin @asciidisco! Chrome can't do 5.1 sound but Kodi can!

**Comment by dagwieers** [comment]
There was a bug in **inpustream.adaptive** where if you selected 720p, you would get a lower quality stream. And if you selected 1080p you would get the 720p stream. It is fixed in peak3d/inputstream.adaptive#259
